src/order/order.d.ts

```diff
@@ -0,0 +1,91 @@
+import type {
```

Comment:
Let's use this file for sharing types that are needed by multiple files. For example, `OrderErrorResponse`, `OrderRequestHeaders`, and `HttpErrorResponse` make sense to keep it here because they are shared by multiple files. The rest of these types should be moved back to the single file they are used in.

Reply:
@xuanwan05 tagging you here in case you missed the above comment. After addressing this we should be good to merge.

Reply:
PatchOrderResponse and PatchRequest types are used by both order-controller and patch-order files since patchOrder api and onShippingChange callback return same PatchOrderResponse type.

The success response status codes of  create order and capture order are 200 and 201, should we create a type which can use by both of the calls?

Reply:
>PatchOrderResponse and PatchRequest types are used by both order-controller and patch-order files since patchOrder api and onShippingChange callback return same PatchOrderResponse type. Should we keep these in the type file?

I'm thinking the Order Controller can import the function as well as the types from the same file for patch.order.ts.

> The success response status codes of create order and capture order are the same (200 and 201), should we create a type which can use by both of the calls?

That's a great point. If create and capture endpoints use the same exact success status codes we can create a shared type for them and have it live in order.d.ts.